### PR TITLE
implement butterknife and view holder for adapter

### PR DESCRIPTION
**Pull request by snufflesrea**
Tried to fix issue #85 issue #95 for Hotels module
Implemented butterknife on any "findViewbyId" and implemented view holder on base adapter
Reformat the coding.

This is the first time I use butterknife, hope it is correct.

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java [hunk 1]
```diff
@@ -131,19 +132,45 @@ public void onDateSet(DatePickerDialog datePickerDialog, int year, int month, in
...
-            case 5: monthString = "May";        break;
-            case 6: monthString = "June";       break;
-            case 7: monthString = "July";       break;
-            case 8: monthString = "August";     break;
-            case 9: monthString = "September";  break;
-            case 10:monthString = "October";    break;
-            case 11:monthString = "November";   break;
-            case 12:monthString = "December";   break;
-            default:monthString = "Invalid month"; break;
+            case 1:
```
Comment (prabhakar267) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java]: I think you can use this [stackoverflow answer](https://stackoverflow.com/q/5799140/5602759)

File: Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java [hunk 2]
```diff
@@ -330,22 +373,25 @@ public void onClick(View view) {
...
+            TextView title;
+            @BindView(R.id.VideoDescription)
+            TextView description;
+            @BindView(R.id.call)
+            LinearLayout call;
+            @BindView(R.id.map)
+            LinearLayout map;
+            @BindView(R.id.book)
+            LinearLayout book;
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java]: Remove this extra line

File: Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java [hunk 3]
```diff
@@ -330,22 +373,25 @@ public void onClick(View view) {
...
-                startActivity(i);
-                break;
-            case R.id.seldate :
-                mDatePickerDialog.setVibrate(isVibrate());
-                mDatePickerDialog.setYearRange(1985, 2028);
-                mDatePickerDialog.setCloseOnSingleTapDay(isCloseOnSingleTapDay());
-                mDatePickerDialog.show(getSupportFragmentManager(), DATEPICKER_TAG);
-                break;
+        class ViewHolder {
+            @BindView(R.id.VideoTitle)
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java]: Change the `id` [here](https://github.com/project-travel-mate/Travel-Mate/blob/master/Android/app/src/main/res/layout/hotel_listitem.xml#L30) to `hotel_name`

File: Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java [hunk 4]
```diff
@@ -330,22 +373,25 @@ public void onClick(View view) {
...
+            TextView title;
+            @BindView(R.id.VideoDescription)
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java]: Change the id name to `hotel_address`.
Change it here: https://github.com/project-travel-mate/Travel-Mate/blob/master/Android/app/src/main/res/layout/hotel_listitem.xml#L62

**Comment by Swati4star** [comment]
Also, there are merge conflicts.

File: Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java (hunk 2 above)
Comment (snufflesrea) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/Hotels.java]: done

**Comment by Swati4star** [comment]
Changes look good now. Squash them into one commit.

**Comment by Swati4star** [comment]
@snufflesrea
There are 105 file changes. You have merged other's changes also in this.

**Comment by Swati4star** [comment]
I am closing this PR. You'll have to open up another PR.

**Pull request closed without merging**